Makefile

```diff
@@ -1,13 +1,76 @@
-.PHONY: all test build run
+include golang.mk
+.DEFAULT_GOAL := test # override default goal set in library makefile
+
 SHELL := /bin/bash
+JAR_DIR := jars
+PKG := github.com/Clever/kinesis-to-firehose
+.PHONY: download_jars run build
 
-all: test build
+URL_PREFIX := http://search.maven.org/remotecontent?filepath=
 
-test:
-	echo "TODO test app"
+# this list lifted from https://github.com/awslabs/amazon-kinesis-client-python/blob/fb49c6390c0593fbcf81d6c34c5245726c15b2f3/setup.py#L60
+JARS_TO_DOWNLOAD := $(addprefix $(JAR_DIR)/,com/amazonaws/amazon-kinesis-client/1.7.2/amazon-kinesis-client-1.7.2.jar \
+com/amazonaws/aws-java-sdk-dynamodb/1.11.14/aws-java-sdk-dynamodb-1.11.14.jar \
+com/amazonaws/aws-java-sdk-s3/1.11.14/aws-java-sdk-s3-1.11.14.jar \
+com/amazonaws/aws-java-sdk-kms/1.11.14/aws-java-sdk-kms-1.11.14.jar \
+com/amazonaws/aws-java-sdk-core/1.11.14/aws-java-sdk-core-1.11.14.jar \
+commons-logging/commons-logging/1.1.3/commons-logging-1.1.3.jar \
+org/apache/httpcomponents/httpclient/4.5.2/httpclient-4.5.2.jar \
+org/apache/httpcomponents/httpcore/4.4.4/httpcore-4.4.4.jar \
+commons-codec/commons-codec/1.9/commons-codec-1.9.jar \
+com/fasterxml/jackson/core/jackson-databind/2.6.6/jackson-databind-2.6.6.jar \
+com/fasterxml/jackson/core/jackson-annotations/2.6.0/jackson-annotations-2.6.0.jar \
+com/fasterxml/jackson/core/jackson-core/2.6.6/jackson-core-2.6.6.jar \
+com/fasterxml/jackson/dataformat/jackson-dataformat-cbor/2.6.6/jackson-dataformat-cbor-2.6.6.jar \
+joda-time/joda-time/2.8.1/joda-time-2.8.1.jar \
+com/amazonaws/aws-java-sdk-kinesis/1.11.14/aws-java-sdk-kinesis-1.11.14.jar \
+com/amazonaws/aws-java-sdk-cloudwatch/1.11.14/aws-java-sdk-cloudwatch-1.11.14.jar \
+com/google/guava/guava/18.0/guava-18.0.jar \
+com/google/protobuf/protobuf-java/2.6.1/protobuf-java-2.6.1.jar \
+commons-lang/commons-lang/2.6/commons-lang-2.6.jar)
+
+EMPTY :=
+SPACE := $(EMPTY) $(EMPTY)
+JAVA_CLASS_PATH := $(subst $(SPACE),:,$(JARS_TO_DOWNLOAD))
+
+$(JARS_TO_DOWNLOAD):
+	mkdir -p `dirname $@`
+	curl -s -L -o $@ -O $(URL_PREFIX)`echo $@ | sed 's/$(JAR_DIR)\///g'`
+
+download_jars: $(JARS_TO_DOWNLOAD)
 
 build:
-	echo "TODO build app"
+	CGO_ENABLED=0 go build -installsuffix cgo -o build/consumer $(PKG)/cmd/consumer
+
+$(GOPATH)/bin/glide:
+	@go get github.com/Masterminds/glide
+
+install_deps: $(GOPATH)/bin/glide
+	@$(GOPATH)/bin/glide install
 
-run: build
-	echo "TODO run app"
+
+KINESIS_STREAM_NAME ?= logs-test
+KINESIS_AWS_REGION ?= us-west-1
+
+# sets the `applicationName` in KCL consume.properties.
+# we want a dif application in Dev and Prod, so that they don't conflict
+# (e.g. app name determines the DynamoDB table that KCL uses to coordinate)
+APPLICATION_NAME ?= kinesis-to-firehose-dev
```

Comment:
excellent point! updated to pass this as an env var and will update ark-config

b9bf9c10576ccdab4037a64f4f34023e895fa215